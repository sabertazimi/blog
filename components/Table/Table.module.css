.table {
  @apply mt-0 mb-9 w-fit border-collapse overflow-auto rounded-lg;

  border-spacing: 0;
}

.table th,
.table td {
  @apply py-2 px-4 text-xl font-normal leading-loose;
}

.table thead tr {
  @apply bg-primary font-semibold text-light;
}

.table th:first-child {
  @apply rounded-tl-lg;
}

.table th:last-child {
  @apply rounded-tr-lg;
}

.table tbody tr {
  @apply border-t border-solid border-primary bg-dark text-light;
}

.table tbody tr:nth-child(2n) {
  @apply bg-gray-800 text-light;
}

.table tbody tr:last-child td:first-child {
  @apply rounded-bl-lg;
}

.table tbody tr:last-child td:last-child {
  @apply rounded-br-lg;
}
