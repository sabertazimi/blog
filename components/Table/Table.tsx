import { cx } from '@components/utils'
import type { HTMLProps } from 'react'
import styles from './Table.module.css'

interface Props extends HTMLProps<HTMLTableElement> {}

function Table({ children, className, ...props }: Props): JSX.Element {
  return (
    <table
      {...props}
      className={cx(
        className,
        styles.table,
        'dark:shadow-xl dark:shadow-primary',
      )}
    >
      {children}
    </table>
  )
}

export default Table
