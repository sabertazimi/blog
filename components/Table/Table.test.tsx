import { render } from '@utils'
import Table from './Table'

describe('table', () => {
  it('should render correctly (snapshot)', () => {
    const { container } = render(<Table />)

    expect(container).toMatchSnapshot()
  })
})
