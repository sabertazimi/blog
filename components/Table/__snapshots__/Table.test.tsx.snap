// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table should render correctly (snapshot) 1`] = `
<div>
  <table
    class="table dark:shadow-xl dark:shadow-primary"
  />
</div>
`;
